backends/ent/retrieve.go

```diff
@@ -0,0 +1,258 @@
+// --------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 The Protobom Authors
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// --------------------------------------------------------------
+package ent
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/protobom/protobom/pkg/sbom"
+	"github.com/protobom/protobom/pkg/storage"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/protobom/storage/internal/backends/ent"
+	"github.com/protobom/storage/internal/backends/ent/document"
+	"github.com/protobom/storage/internal/backends/ent/metadata"
+)
+
+// Retrieve implements the storage.Retriever interface.
+func (backend *Backend) Retrieve(id string, _opts *storage.RetrieveOptions) (*sbom.Document, error) {
+	if backend.BackendOptions == nil {
+		backend.BackendOptions = NewBackendOptions()
+	}
+
+	backend.init(backend.BackendOptions)
+
+	defer backend.client.Close()
```

Comment:
What's the lifecycle of the client? Are we introducing an inefficiency here by initializing and closing it on every `Store()` and `Retrieve()` operation?

Reply:
Open to ideas about how to guarantee the database gets closed when finished, it's currently preventing use of an in-memory database as well

Reply:
See https://github.com/protobom/storage/pull/4#issuecomment-2147895186 for how this was reworked